### ISSUE TITLE
A11y Issue: Keyboard user is not able to access the focusable control present in the ng-select MenuOption

### ISSUE THREAD

**Issue by sachinkkarande** [ng-select/ng-select#2284]
**Describe the bug**
A11y Issue: Keyboard user is not able to access the focusable control present in the MenuOption
Keyboard focus is not going inside the MenuOption if the Menu option contains a Focusable element like Button/Tooltip Icon.

This is Accessibility issue since the Actionable control inside MenuOption is only accessible through the Mouse and not with the keyboard.

**Reproducbile example**
Refer https://stackblitz.com/edit/ng-select-option-tooltip?file=src%2Fapp%2Fapp.component.html

In the example, Inside MenuOption there is an info icon having tooltip.
Try using only the keyboard, and observe that the keyboard user is not able to focus on the info icon and the user is not able to see the tooltip. 
Observe that ToolTips are visible for users using Mouse.

This is a kind of blocker from an Accessibility perspective for keyboard-only users.

**Expected behavior**
Keyboard only users should be able to navigate to actionable HTML controls(Like buttons) inside MenuOptions. 

**Screenshots**
Refer "ng-select-menu-option-having-actionable-control.png"
<img width="1792" alt="ng-select-menu-option-having-actionable-control" src="https://github.com/ng-select/ng-select/assets/83701351/808707aa-91a1-4904-a1f7-8d1d1a813ba4">

Also refer "ng-select-MeuOption-having-Focusable-item-not-keyboard-accessible.gif".
![ng-select-MeuOption-having-Focusable-item-not-keyboard-accessible](https://github.com/ng-select/ng-select/assets/83701351/f09bd0df-50de-46d5-bba6-6f6a90d083ea)

**Desktop (please complete the following information):**
 - OS: Not related to any specific OS.
 - Browser: Not related to any specific Browser.
 - Version Not related to any specific Version.

**Additional context**
This is Accessibility Issue.
